tiatoolbox/tools/registration/wsi_registration.py

```diff
@@ -0,0 +1,68 @@
+import cv2
+import numpy as np
+import torch
+import torchvision
+from torchvision.models._utils import IntermediateLayerGetter
+
+
+class DFBRegistrtation:
+    r"""Deep Feature based Registration
+
+    This class implements a CNN feature based registration,
+    as proposed in a paper titled `Deep Feature based Cross-slide Registration
+    <https://arxiv.org/pdf/2202.09971.pdf>`_.
+
+    """
+
+    def __init__(self):
+        self.patch_size = (224, 224)
+        self.Xscale, self.Yscale = [], []
+        model = torchvision.models.vgg16(True)
+        return_layers = {"16": "block3_pool", "23": "block4_pool", "30": "block5_pool"}
+        self.FeatureExtractor = IntermediateLayerGetter(
+            model.features, return_layers=return_layers
+        )
+
```

Comment:
```suggestion
    # Make this function private when full pipeline is implemented.
```

Reply:
OK

Reply:
Please put this as a comment. It can be removed once this has been done.